jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -472,14 +472,25 @@ ecma_create_property (ecma_object_t *object_p, /**< the object */
 
   first_property_pair_p->values[1] = value;
 
-  /* See the comment before the other ecma_property_hashmap_insert above. */
+  /* Need to recheck whether the property hashmap of the object exists, bacuse the
+   * allocation of the new property hashmap pair may have deleted it. */
 
-  if (has_hashmap && name_p != NULL)
+  property_list_head_p = &object_p->property_list_or_bound_object_cp;
```

Comment:
Please delete the has_hashmap boolean value and do a recheck in the code above (line 447) as well.